src/Shell32.Desktop/Shell32+SHITEMID.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+
+namespace PInvoke
+{
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="SHITEMID"/> nested type.
+    /// </content>
+    public partial class Shell32
+    {
+        /// <summary>
+        /// Defines an item identifier.
+        /// </summary>
+        /// <remarks>Used by <see cref="ITEMIDLIST"/></remarks>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SHITEMID
+        {
+            /// <summary>
+            /// The size of identifier, in bytes, including cb itself.
+            /// </summary>
+            ushort cb;
+
+            /// <summary>
+            /// A variable-length item identifier.
+            /// </summary>
+            public byte[] abID;
```

Comment:
Ah yes this isn't doable with the marshaler anyway. it should be done by declaring a fixed 1 element array and then allocating the structure by hand if we need to allocate it (Exactly as the C++ code does)

I don't know if we need to get values in and out of this structure (Maybe it's declared but enough API are available that it's never needed to access it). If it is necessary it would look like that :

```csharp
public unsafe partial class Shell32
{
	/// <summary>
	/// Defines an item identifier.
	/// </summary>
	/// <remarks>Used by <see cref="ITEMIDLIST"/></remarks>
	[StructLayout(LayoutKind.Sequential)]
	public struct SHITEMID
	{
		public const short FixedPartSize = 2; // sizeof(ushort)
		
		/// <summary>
		/// The size of identifier, in bytes, including cb itself.
		/// </summary>
		public short cb;

		/// <summary>
		/// A variable-length item identifier.
		/// </summary>
		public fixed byte abID[1];

		public byte[] GetValue()
		{
			var length = cb - FixedPartSize;
			var result = new byte[length];
			fixed (byte* valuePtr = abID)
			{
				Marshal.Copy((IntPtr)valuePtr, result, 0, length);
			}
			return result;
		}

		public static SHITEMID* Create(short size)
		{
			var totalSize = FixedPartSize + size;
			var result = (SHITEMID*)Marshal.AllocHGlobal(totalSize);
			result->cb = (short)totalSize;
			return result;
		}
	}
}
```